server/service/integration_enterprise_test.go

```diff
@@ -16834,6 +16834,16 @@ func (s *integrationEnterpriseTestSuite) TestListHostSoftwareWithLabelScoping()
 	s.DoJSON("GET", fmt.Sprintf("/api/latest/fleet/hosts/%d/software", host.ID), nil, http.StatusOK, &getHostSw)
 	require.Len(t, getHostSw.Software, 1)
 
+	// installer should be out of scope since the label is "exclude any"
+	scoped, err := s.ds.IsSoftwareInstallerLabelScoped(ctx, installerID, host.ID)
+	require.NoError(t, err)
+	require.False(t, scoped)
+
+	// host should be excluded from the software installer
+	hostsNotInScope, err := s.ds.GetExcludedHostIDMapForSoftwareInstaller(ctx, installerID)
+	require.NoError(t, err)
+	require.Contains(t, hostsNotInScope, host.ID)
+
 	// uninstall the software
 	s.DoJSON("POST", fmt.Sprintf("/api/latest/fleet/hosts/%d/software/%d/uninstall", host.ID, titleID), nil, http.StatusAccepted, &resp)
 	s.DoJSON("GET", fmt.Sprintf("/api/latest/fleet/hosts/%d/software", host.ID), nil, http.StatusOK, &getHostSw)
```

Comment:
This test is failing because it is expecting a `SoftwarePackage` to be present.
`getHostSw.Software[0].SoftwarePackage.LastInstall` (throws a nil pointer since `SoftwarePackage` is nil, and fails)

However, as you can see above here the installer is excluded and so is the host via label scoping. 

If I am understanding the original logic correctly it has this weird statement where scoping is ignored if
```
lsua.created_at > lsia.created_at AND hsr.exit_code = 0
```
https://github.com/fleetdm/fleet/blob/rc-minor-fleet-v4.67.0/server/datastore/mysql/software.go#L2696C38-L2696C93

Why are we ignore labels in this case?

Reply:
My best guess here is that the original impl was attempting to treat uninstalls (and maybe installs) as more authoritative because once they complete the status is reported instantly, where inventory runs off of host vitals. Idea being that if software was successfully uninstalled (exit code 0) and the latest uninstall happened after the latest install, _omitting_ the info we have about uninstalls would cause the software to show up until inventory had a chance to catch up, which would look bugged if someone uninstalled the software, then dropped the host from scope.

We wouldn't have this issue if inventory was recalculated right after software (un)installs, which is in the same realm as #28738, but we have these weird edge cases until then.

Reply:
With the above said, given that we have to show software packages in order to show software install timestamps based on how that data structure works, thinking we revise the test, comment with a TODO for now that indicates we need to understand/clean up this behavior (which we'll do at least with installer scoping changes), and move on.

Reply:
Per discussion today with @mostlikelee, if an installer is out of scope, we treat it consistently as if it doesn't exist, so inventory gets used as the only source for existence checks/versions.